controllers/logging/logging_controller.go

```diff
@@ -274,10 +285,53 @@ func (r *LoggingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 			return *result, err
 		}
 	}
+	if logging.Status.FluentdConfigName != "" {
+		return r.checkFluentdConfigFinalizer(ctx, &logging, log)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *LoggingReconciler) checkFluentdConfigFinalizer(ctx context.Context, logging *loggingv1beta1.Logging, log logr.Logger) (ctrl.Result, error) {
+	fluentdConfigFinalizer := "fluentdconfig.logging.banzaicloud.io/finalizer"
 
+	if logging.DeletionTimestamp.IsZero() {
+		if !controllerutil.ContainsFinalizer(logging, fluentdConfigFinalizer) {
+			controllerutil.AddFinalizer(logging, fluentdConfigFinalizer)
+			if err := r.Update(ctx, logging); err != nil {
+				return ctrl.Result{}, nil
+			}
+		}
+	} else {
+		// Marked for deletion, check for fluentdConfig
+		fluentdConfigName := types.NamespacedName{Namespace: logging.Spec.ControlNamespace, Name: logging.Status.FluentdConfigName}
+		if controllerutil.ContainsFinalizer(logging, fluentdConfigFinalizer) && r.isExistingFluentdConfig(ctx, fluentdConfigName) {
+			return reconcile.Result{}, errors.NewWithDetails("failed to delete logging resources, delete fluentdConfig first", "fluentdConfig", logging.Status.FluentdConfigName)
```

Comment:
I'm thinking about this hard because the user won't have any feedback about this. How about one of the following?
- we could actually remove the fluentdconfig resource here
- we could update the logging status so that the user see what's happening

or maybe it would be enough to use a logging ownerref on the fluentdconfig?

Reply:
We can not update the status once an object is marked for deletion: https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/#how-finalizers-work (see the note section)

Removing the fluentdconfig might be unexpected (admin deletes logging, tenant loses it's fluentdconfig object)